File: Leetcode/PythonAnswers/Easy/TransformedArray.py
#Leetcode Problem 3379 Transformed Array
from typing import List

class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        #Itereate through the list and append the new value to the result list.
        result = []
        for i in range(len(nums)):
            n= nums[i]
            if i + n < len(nums) and i + n >= 0:
                result.append(nums[i + n])
            elif i + n >= len(nums):
                overflow = (i + n) // len(nums)
                overflow_result = (i+n) - (overflow * len(nums))
                result.append(nums[overflow_result])
            else:
                underflow = (i + n) // len(nums)
                underflow_result = (i+n) - (underflow * len(nums))
                result.append(nums[underflow_result])
        return result

#Accepted answer.

File: Leetcode/Java/Easy/TransformedArray.java
//Accepted answer for leetcode problem 3379 Transformed Array

class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int[] result = new int[nums.length];

        for (int i = 0; i <nums.length; i++){
            int n = nums[i];
            if (i + n < nums.length && i + n >= 0){
                result[i] += nums[i +n];
            } else if (i + n >=nums.length){
               int overflow = (i + n) / nums.length;
               int overflow_result = (i + n) -(overflow * nums.length);
               result[i] += nums[overflow_result];
            }else{
                int underflow = (i +n) / nums.length;
                int underflow_result = (i + n) - (underflow * nums.length);
                if (underflow_result < 0){
                    underflow_result += nums.length;
                    result[i] += nums[underflow_result];
                }else{
                result[i] += nums[underflow_result];
            }
        }        
        }
    return result;
    }
}